hangman/lib/hangman/game.ex

```diff
@@ -0,0 +1,43 @@
+defmodule Hangman.Game do
+  @moduledoc """
+  This module contains the implementation of the Hangman API
+  """
+  import Hangman.Util
+
+  defstruct turns_left: 7, game_state: :initializing,
+            word: "", used: [], last_guess: '', letters: []
+
+  @doc """
+  Returns the initial state for a game
+  """
+  def new_game do
+    word = Dictionary.random_word()
+    letters = for _ <- String.codepoints(word), do: "-"
+    %Hangman.Game{}
+    |> Map.put(:word, word)
+    |> Map.put(:letters, letters)
+  end
+
+  @doc """
+  Returns the current tally of a game, which is essentially the
+  state, with the word hidden
+  """
+  def tally(state) do
```

Comment:
## -1

Simply returning the internal state, minus the word, is a little dangerous. It works for now, but in the real world someone coming along later could add a field to the state and not realize that it would by default be copied out through the API.